### PR TITLE
VitePress migration

### PR DESCRIPTION
**Pull request by lifehackerhansol**
**Description**

In line with the other hacks-guide migrations, I have applied the same migration here.

Currently buttons do not work as well as it did on mkdocs, so this was switched to an admonition, similar to how the 3DS guide operates. There doesn't seem to be a much better option.

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -3,53 +3,52 @@ name: Deploy site
 on:
```
Comment (suchmememanyskill) [inline .github/workflows/publish.yml:3]: (Unrelated to this file)
Do you think it's worth having a CI build check linting, so after you fixing stuff it stays consistent?

File: docs/extras/autorcm.md [hunk 2]
```diff
@@ -1,17 +1,22 @@
...
-    You really want to kill your console, huh? If you haven't made a BOOT0/1 backup yet, it is recommended to make one **right now**.
+::: warning
 
-    1. Boot into Hekate
-    1. Go to `Tools`, and select `Backup eMMC`
-    1. Then tap `eMMC BOOT0 & BOOT1` and let the process complete.
+**If you don't have a BOOT0/1 backup yet...**
 
-### **Disadvantages of AutoRCM**
+You really want to kill your console, huh? If you haven't made a BOOT0/1 backup yet, it is recommended to make one **right now**.
```
Comment (suchmememanyskill) [inline docs/extras/autorcm.md:11]: Unrelated to this PR, but we may want to check for such snarky comments in the future, and remove them.

File: docs/extras/updating.md [hunk 3]
```diff
@@ -4,12 +4,17 @@ This page documents how you can keep your system up-to-date.
...
-### **Updating Atmosphere**
+## Updating Atmosphere
 
 When updating Atmosphère, always make sure to _read the release notes_. They may list important changes and modifications to your system.
 
-!!! warning "Updating from below Atmosphère 1.0.0"
-    If you update from below Atmosphère 1.0.0, there are additional steps to follow. You will have to delete the `sept` folder from your microSD, delete `fusee-secondary.bin` from your `atmosphere` folder and update your Hekate config file: <a href="../../files/emu/hekate_ipl.ini" download>hekate_ipl.ini</a> in the `bootloader` folder.
+::: warning
+
+**Updating from below Atmosphère 1.0.0**
```
Comment (suchmememanyskill) [inline docs/extras/updating.md:13]: Unrelated to this PR, but we're on atmosphere 1.8.0. I don't expect anyone to still be on something below 1.0.0.

File: docs/homebrew/nx-shell.md [hunk 4]
```diff
@@ -1,9 +1,8 @@
-### **NX-Shell**
-NX-Shell is a file manager.
+# NX-Shell
 
------
+NX-Shell is a file manager.
```
Comment (suchmememanyskill) [inline docs/homebrew/nx-shell.md:3]: Unrelated to this PR, but probably should expand this to 'NX-Shell is an on-console file manager, able to be launched from the homebrew menu'. Such a short introduction sounds odd to me.

File: docs/extras/updating.md [hunk 5]
```diff
@@ -4,12 +4,17 @@ This page documents how you can keep your system up-to-date.
 
 After following our guide, your system will consist of three core elements that can be updated. Atmosphere, Hekate and your system firmware.
 
-### **Updating Atmosphere**
+## Updating Atmosphere
```
Comment (suchmememanyskill) [inline docs/extras/updating.md:7]: Unrelated to this PR, but we should probably expand this page to feature common errors, and additional steps users may have to take (like updating/removing sysmodules).

May also be worth splitting this page off in 2 sepearte pages, one to update the firmware, one to update cfw.

File: docs/index.md [hunk 6]
```diff
@@ -1,51 +1,47 @@
-<a href="https://discord.gg/C29hYvh"><img style="float: right;" src="img/discord.png"></a>
-
-# **NH Switch Guide**
-
-The NH Switch Guide is a collaboration from Nintendo Homebrew's Discord community, getting you from a stock Switch to Atmosphère.
-
-For live support with this guide, visit us in **#switch-assistance-1** or **#switch-assistance-2** over at the [NH-Discord server](https://discord.gg/C29hYvh).
+---
+layout: home
```
Comment (suchmememanyskill) [inline docs/index.md:2]: Where are these layouts defined?

File: docs/user_guide/getting_started.md [hunk 7]
```diff
@@ -49,50 +52,70 @@ The following information is based on [this GBATemp thread](https://gbatemp.net/
...
-| 8.0.1            | [**RCM**](rcm/index.md)               | [**Modchip**](modchip/index.md) (no software exploit)          | [**Modchip**](modchip/index.md) (no software exploit)| [**Modchip**](modchip/index.md) (no software exploit)|
-| 8.1.0 and up     | [**RCM**](rcm/index.md)               | [**Modchip**](modchip/index.md) (no software exploit)          | [**Modchip**](modchip/index.md) (no software exploit)| [**Modchip**](modchip/index.md) (no software exploit)|
+| 1.0.0            | Nereba or [**RCM**](rcm/index)     | **N/A**                          | **N/A**                     | **N/A**                                              |
+| 2.0.0 - 3.0.2    | Caffeine or [**RCM**](rcm/index)   | **N/A**                          | **N/A**                     | **N/A**                                              |
+| 4.0.0 - 4.1.0    | Caffeine or [**RCM**](rcm/index)   | Caffeine                         | **N/A**                     | **N/A**                                              |
+| 5.0.0 - 7.0.0    | [**RCM**](rcm/index)               | [**Modchip**](modchip/index) / Wait for CFW                 | **N/A**                     | **N/A**                |
```
Comment (suchmememanyskill) [inline docs/user_guide/getting_started.md:70]: I doubt this will ever happen lol

**Review by suchmememanyskill — COMMENTED** [review]
Haven't scrolled through the web preview yet, but i see nothing weird changed in the markdown. Well done.

File: docs/index.md (hunk 6 above)
Comment (lifehackerhansol) [reply docs/index.md:2]: It's built into VitePress. https://vitepress.dev/reference/default-theme-layout

File: .github/workflows/publish.yml (hunk 1 above)
Comment (lifehackerhansol) [reply .github/workflows/publish.yml:3]: I don't know of any particular one; would be nice to have though.

**Pull request merged**